Build failure: nixpkgs.rPackages.sf

### Steps To Reproduce

Steps to reproduce the behavior:

```
{ nixpkgs ? import <nixpkgs> { } }:

let
  R-with-my-packages = nixpkgs.rWrapper.override{
    packages = with nixpkgs.rPackages; [
      sf
    ]; };

in

nixpkgs.stdenv.mkDerivation {
  name = "rOnly";

  buildInputs = [
    R-with-my-packages
  ];
}
```

```
nix-shell -I nixpkgs=/Users/dom/nixpkgs rshell.nix
```

### Build log

```
nix log /nix/store/nwj1ggh3k9q7yf6b5h5jlk85nrjy7gbh-r-sf-1.0-14.drv

@nix { "action": "setPhase", "phase": "unpackPhase" }
unpacking sources
unpacking source archive /nix/store/v7m926inrsbxxs0bqhbagq5ba7rffixq-sf_1.0-14.tar.gz
source root is sf
setting SOURCE_DATE_EPOCH to timestamp 1689064802 of file sf/MD5
@nix { "action": "setPhase", "phase": "patchPhase" }
patching sources
@nix { "action": "setPhase", "phase": "updateAutotoolsGnuConfigScriptsPhase" }
updateAutotoolsGnuConfigScriptsPhase
@nix { "action": "setPhase", "phase": "configurePhase" }
configuring
@nix { "action": "setPhase", "phase": "buildPhase" }
building
@nix { "action": "setPhase", "phase": "checkPhase" }
running tests
@nix { "action": "setPhase", "phase": "installPhase" }
installing
 * installing *source* package ‘sf’ ...
 ** package ‘sf’ successfully unpacked and MD5 sums checked
 ** using staged installation
configure: CC: clang
configure: CXX: /nix/store/lhqd2mwnd8wf1zh4jrd73hp8z1v5q9jw-clang-wrapper-11.1.0/bin/c++ -std=gnu++17
checking for gdal-config... /nix/store/wfm8xa3qz9l3jdzs9a6dnx73y7lklg53-gdal-3.7.2/bin/gdal-config
checking gdal-config usability... yes
configure: GDAL: 3.7.2
checking GDAL version >= 2.0.1... yes
checking for gcc... clang
checking whether the C compiler works... yes
checking for C compiler default output file name... a.out
checking for suffix of executables... 
checking whether we are cross compiling... no
checking for suffix of object files... o
checking whether the compiler supports GNU C... yes
checking whether clang accepts -g... yes
checking for clang option to enable C11 features... none needed
checking for stdio.h... yes
checking for stdlib.h... yes
checking for string.h... yes
checking for inttypes.h... yes
checking for stdint.h... yes
checking for strings.h... yes
checking for sys/stat.h... yes
checking for sys/types.h... yes
checking for unistd.h... yes
checking for gdal.h... yes
checking GDAL: linking with --libs only... yes
checking GDAL: /nix/store/wfm8xa3qz9l3jdzs9a6dnx73y7lklg53-gdal-3.7.2/share/gdal/pcs.csv readable... no
checking GDAL: checking whether PROJ is available for linking:... yes
checking GDAL: checking whether PROJ is available for running:... yes
configure: GDAL: 3.7.2
configure: pkg-config proj exists, will use it
configure: using proj.h.
Package libtiff-4 was not found in the pkg-config search path.
Perhaps you should add the directory containing `libtiff-4.pc'
to the PKG_CONFIG_PATH environment variable
Package 'libtiff-4', required by 'proj', not found
configure: PROJ: 9.3.0
checking PROJ: checking whether PROJ and sqlite3 are available for linking:... no
configure: error: libproj or sqlite3 not found in standard or given locations.
ERROR: configuration failed for package ‘sf’
 * removing ‘/nix/store/f1ywv3w1q5mdpwvr28xqb2y005hk7vc3-r-sf-1.0-14/library/sf’
```

### Additional context

```
git log --oneline -10

d83248b4c9db (HEAD -> reallyreallymaster, origin/master, origin/HEAD) Merge pull request #257012 from loco-choco/owmods-clia175fdabc1b3 Merge pull request #256416 from rhysmdnz/edge-updates
3bdc58732de0 Merge pull request #256741 from marcusramberg/marcus/fix_streamdeck_ui
c463204b608a Merge pull request #256858 from r-ryantm/auto-update/hop-cli
2b9a9254e327 Merge pull request #257002 from mweinelt/ctranslate-tensorflow-bin
5765812aaa29 Merge pull request #256997 from aaronjheng/mongosh
05698bba1eba Merge pull request #256744 from nadir-yttrium/update-clifm-1.14.6
ade134ecd189 nixpkgs manual: doc python: add hyperlinks to Python section (#252156)
8352a244b20d Merge pull request #256905 from hulr/setzer
e9c44cb20358 librepcb: 0.1.7 -> 1.0.0
```

### Notify maintainers

@jbedo 

### Metadata

Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.

```console
[user@system:~]$ nix-shell -p nix-info --run "nix-info -m"

nix-shell -p nix-info --run "nix-info -m"
these 2 paths will be fetched (0.01 MiB download, 0.11 MiB unpacked):
  /nix/store/0025zncyzwcmsp3v3z1srs51b7py6psc-nix-info
  /nix/store/82w7k2rr446hjhi29kd8ckbpgbiykpiw-DarwinTools-1
copying path '/nix/store/82w7k2rr446hjhi29kd8ckbpgbiykpiw-DarwinTools-1' from 'https://cache.nixos.org'...
copying path '/nix/store/0025zncyzwcmsp3v3z1srs51b7py6psc-nix-info' from 'https://cache.nixos.org'...
 - system: `"aarch64-darwin"`
 - host os: `Darwin 21.6.0, macOS 12.6`
 - multi-user?: `yes`
 - sandbox: `no`
 - version: `nix-env (Nix) 2.13.3`
 - channels(root): `"nixpkgs"`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixpkgs`
```

--- a/pkgs/development/r-modules/default.nix
+++ b/pkgs/development/r-modules/default.nix
@@ -423,7 +423,7 @@ let
     seqinr = [ pkgs.zlib.dev ];
     webp = [ pkgs.pkg-config ];
     seqminer = with pkgs; [ zlib.dev bzip2 ];
-    sf = with pkgs; [ gdal proj geos ];
+    sf = with pkgs; [ gdal proj geos libtiff curl ];
     terra = with pkgs; [ gdal proj geos ];
     showtext = with pkgs; [ zlib libpng icu freetype.dev ];
     simplexreg = [ pkgs.gsl ];

